pkg/api/dryrunclient.go

```diff
@@ -155,6 +155,17 @@ func (d *DryRunClient) VolumeRemove(ctx context.Context, volumeID string, force
 	return ErrNotImplemented
 }
 
+func (d *DryRunClient) ContainerExecCreate(ctx context.Context, container string, config moby.ExecConfig) (moby.IDResponse, error) {
+	fmt.Printf("%sCreating Exec configuration for container %s with command '%s'\n", DRYRUN_PREFIX, container, strings.Join(config.Cmd, " "))
+	config.Cmd = []string{"true"}
+	return d.apiClient.ContainerExecCreate(ctx, container, config)
+}
+
+func (d *DryRunClient) ContainerExecStart(ctx context.Context, execID string, config moby.ExecStartCheck) error {
+	fmt.Printf("%sExecuting command in detach mode\n", DRYRUN_PREFIX)
```

Comment:
why "in detach(ed) mode" ?
when starting exec command, you can't tell if client did/will attach or not

Reply:
When I tested all the potential cases, I identified that `ContainerExecStart` was only called when the detached mode is activated
https://github.com/docker/cli/blob/24b4924410dc1db788ce1ee59e2cc7fa8459f8b0/cli/command/container/exec.go#L139

Reply:
Do you think it's worth having something like `map[string]ExecDetails` on the `DryRunClient`?

That way, `ExecCreate` could be silent, but here we could log out `DRY-RUN Running command "rm -rf /" in app-svc-1`.

(Don't feel strongly about this, also seems fine as-is!)

Reply:
`ContainerExecAttach` in the client indeed calls the `/start` engine endpoint with `Detach` set but this can be used to distinguish detached mode.